packages/react-query/src/__tests__/useQuery.test.tsx

```diff
@@ -6225,4 +6225,24 @@ describe('useQuery', () => {
 
     await waitFor(() => rendered.getByText('Works'))
   })
+
+  // For Project without TS, when migrating from v4 to v5, make sure invalid calls due to bad parameters are tracked.
+  it('should throw in case of bad arguments to enhance DevX', async () => {
+    // Mock console error to avoid noise when test is run
+    const consoleMock = vi
+      .spyOn(console, 'error')
+      .mockImplementation(() => undefined)
+
+    const key = queryKey()
+    const queryFn = () => 'data'
+
+    function Page() {
+      // Invalid call on purpose
+      const { data } = useQuery(key, { queryFn })
```

Comment:
doesn't this need a `@ts-expect-error` ?

Reply:
yeah, there are two errors:

```
src/__tests__/useQuery.test.tsx(6241,13): error TS6133: 'data' is declared but its value is never read.
src/__tests__/useQuery.test.tsx(6241,33): error TS2769: No overload matches this call.
```

Reply:
… not familiar enough with TS… will fix soon.

Reply:
Ok so tests pass locally using: `pnpm test` in root dir and `pnpm test:lib` in packages/react-query dir.
But CI seems to fail…

Any idea?

Reply:
A race condition in useQueries tests?

Reply:
flaky tests :(